app/javascript/components/patient/close_contacts/CloseContactModal.js

```diff
@@ -58,89 +42,136 @@ class CloseContactModal extends React.Component {
+  handleLDEChange = date => {
+    this.setState({ last_date_of_exposure: date });
+  };
 
   handleAssignedUserChange = event => {
     if (isNaN(event.target.value) || parseInt(event.target.value) > 999999) return;
     // trim() call included since there is a bug with yup validation for numbers that allows whitespace entry
     const value = _.trim(event.target.value) === '' ? null : parseInt(event.target.value);
-    this.updateState('assigned_user', value);
+    this.setState({ assigned_user: value });
   };
 
-  updateState(key, value) {
-    this.setState({ [key]: value }, () => {
-      let isValid = (this.state.first_name || this.state.last_name) && (this.state.primary_telephone || this.state.email);
-      this.setState({
-        isValid,
+  /**
+   * Validates close contact data and submits if valid, otherwise displays errors in modal
+   */
+  submit = () => {
+    schema
+      .validate({ ...this.state }, { abortEarly: false })
+      .then(() => {
+        this.setState({ loading: true }, () => {
+          this.props.onSave({
+            assigned_user: this.state.assigned_user,
+            contact_attempts: this.state.contact_attempts,
+            email: this.state.email,
+            first_name: this.state.first_name,
+            last_date_of_exposure: this.state.last_date_of_exposure,
+            last_name: this.state.last_name,
+            notes: this.state.notes,
+            primary_telephone: this.state.primary_telephone,
+          });
+        });
+      })
+      .catch(err => {
+        // Validation errors, update state to display to user
+        if (err && err.inner) {
+          let issues = {};
+          for (const issue of err.inner) {
+            issues[issue['path']] = issue['errors'];
+          }
+          this.setState({ errors: issues });
+        }
       });
-    });
-  }
+  };
 
   render() {
+    // Data is valid as long as at least one field has a value
+    const isValid = (this.state.first_name || this.state.last_name) && (this.state.primary_telephone || this.state.email);
+
     return (
       <Modal size="lg" show centered onHide={this.props.onClose}>
-        <h1 className="sr-only">{this.props.title}</h1>
+        <h1 className="sr-only">{this.props.editMode ? 'Edit' : 'Add New'} Close Contact</h1>
         <Modal.Header>
-          <Modal.Title>{this.props.title}</Modal.Title>
+          <Modal.Title>{this.props.editMode ? 'Edit' : 'Add New'} Close Contact</Modal.Title>
         </Modal.Header>
-        <Modal.Body className="px-5">
-          <Row className="mt-3">
-            <Form.Group as={Col} lg="12" controlId="cc_first_name">
-              <Form.Label className="input-label">First Name {schema?.fields?.first_name?._exclusive?.required && '*'} </Form.Label>
-              <Form.Control size="lg" className="form-square" value={this.state.first_name || ''} onChange={this.handleNameChange} />
+        <Modal.Body>
```

Comment:
Something is now off with the alignment of elements in this modal when it snaps to the low breakpoint

<img width="734" alt="image" src="https://user-images.githubusercontent.com/8235974/149205383-efd0affb-2430-48d0-ab27-a0814be50d60.png">

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1184/commits/8863abf84897976b012bf1be7a5fe4644c92b70d